File: client/src/app/modules/tiers/services/rff.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { Rff } from '../../tiers/model/Rff';
import { RffSearch } from '../model/RffSearch';

@Injectable({
  providedIn: 'root'
})
export class RffService extends AppRestService<Rff>{

  constructor(http: HttpClient) {
    super(http, "tiers");
  }

  getRffs(rffSearch: RffSearch) {
    return this.postList(new HttpParams(), "rffs", rffSearch);
  }

  addOrUpdateRff(rff: Rff) {
    return this.addOrUpdate(new HttpParams(), "rff", rff, "Enregistré", "Erreur lors de l'enregistrement");
  }

  cancelRff(rff: Rff) {
    return this.get(new HttpParams().set("idRff", rff.id), "rff/cancel");
  }

  sendRff(rff: Rff, amount: number, sendToMe: boolean) {
    return this.get(new HttpParams().set("idRff", rff.id).set("amount", amount).set("sendToMe", sendToMe), "rff/send", "Mail envoyé !");
  }

  generateInvoiceForRff(rff: Rff) {
    return this.get(new HttpParams().set("idRff", rff.id), "rff/invoice");
  }

}

File: src/main/java/com/jss/osiris/modules/osiris/tiers/service/RffService.java
package com.jss.osiris.modules.osiris.tiers.service;

import java.util.List;

import com.jss.osiris.libs.exception.OsirisClientMessageException;
import com.jss.osiris.libs.exception.OsirisDuplicateException;
import com.jss.osiris.libs.exception.OsirisException;
import com.jss.osiris.libs.exception.OsirisValidationException;
import com.jss.osiris.modules.osiris.invoicing.model.Invoice;
import com.jss.osiris.modules.osiris.tiers.model.Rff;
import com.jss.osiris.modules.osiris.tiers.model.RffSearch;

public interface RffService {

    public Rff getRff(Integer id);

    public List<Rff> getRffs(RffSearch rffSearch) throws OsirisException;

    public Rff addOrUpdateRff(Rff rff);

    public Rff cancelRff(Rff rff);

    public Invoice generateInvoiceForRff(Rff rff)
            throws OsirisValidationException, OsirisException, OsirisClientMessageException, OsirisDuplicateException;

    public Rff sendRff(Rff rff, Float amount, boolean sendToMe) throws OsirisException, OsirisClientMessageException;

}
